flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -182,4 +219,102 @@ public FileStoreSourceSplitSerializer getSplitSerializer() {
+                    new BinaryRowDataSerializer(fileStore.partitionType().getFieldCount());
+            DataFileMetaSerializer metaSerializer =
+                    new DataFileMetaSerializer(fileStore.keyType(), fileStore.valueType());
+            DataInputViewStreamWrapper view = new DataInputViewStreamWrapper(in);
+            specifiedManifestEntries = new HashMap<>();
+            int partitionCtr = view.readInt();
+            while (partitionCtr > 0) {
+                BinaryRowData partition = partSerializer.deserialize(view);
+                Map<Integer, List<DataFileMeta>> bucketEntry = new HashMap<>();
+                int bucketCtr = view.readInt();
+                while (bucketCtr > 0) {
+                    int bucket = view.readInt();
+                    int entryCtr = view.readInt();
+                    if (entryCtr == 0) {
+                        bucketEntry.put(bucket, Collections.emptyList());
+                    } else {
+                        List<DataFileMeta> metas = new ArrayList<>();
+                        while (entryCtr > 0) {
+                            metas.add(metaSerializer.deserialize(view));
+                            entryCtr--;
+                        }
+                        bucketEntry.put(bucket, metas);
+                    }
+                    bucketCtr--;
+                }
+                specifiedManifestEntries.put(partition, bucketEntry);
+                partitionCtr--;
+            }
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof FileStoreSource)) {
+            return false;
+        }
+        FileStoreSource that = (FileStoreSource) o;
+        return valueCountMode == that.valueCountMode
+                && isContinuous == that.isContinuous
+                && discoveryInterval == that.discoveryInterval
+                && latestContinuous == that.latestContinuous
+                && fileStore.equals(that.fileStore)
+                && Arrays.equals(projectedFields, that.projectedFields)
+                && Objects.equals(partitionPredicate, that.partitionPredicate)
+                && Objects.equals(fieldPredicate, that.fieldPredicate)
+                && Objects.equals(specifiedSnapshotId, that.specifiedSnapshotId)
+                && Objects.equals(specifiedManifestEntries, that.specifiedManifestEntries);
+    }
+
+    @Override
+    public int hashCode() {
+        int result =
+                Objects.hash(
+                        fileStore,
+                        valueCountMode,
+                        isContinuous,
+                        discoveryInterval,
+                        latestContinuous,
+                        partitionPredicate,
+                        fieldPredicate,
+                        specifiedSnapshotId,
+                        specifiedManifestEntries);
+        result = 31 * result + Arrays.hashCode(projectedFields);
```

Comment:
It's recommended to use `Arrays.deepHashCode` for multiple dimension array, which is similar to the above comment.